### PR TITLE
KeyName parameter added

### PR DESCRIPTION
**Pull request by HarryKodden**
Added parameters key_info_name and key_info_cert to sign method

### CODE REVIEW

File: signxml/__init__.py [hunk 1]
```diff
@@ -315,6 +315,13 @@ def sign(self, method=methods.enveloped, algorithm="rsa-sha256", key=None, passp
         :param reference_uri:
             Custom reference URI to incorporate into the signature. Only used when ``method`` is set to ``detached``.
         :type reference_uri: string
+        :param key_info_name:
```
Comment (kislyuk) [inline signxml/__init__.py]: I think this should be called `key_name`.

**Comment by kislyuk** [comment]
Thanks for the PR. Please do the following before we can merge:
- Eliminate the `key_info_cert` input parameter. There is no reason for it.
- Rename the `key_info_name` parameter to `key_info`.
- Expand the description to reference the spec content at  http://www.w3.org/TR/xmldsig-core2/#sec-KeyName and paste a description of what the key name means from there.

**Comment by hk0** [comment]
Reason for including key_info_cert boolean parameter is that i need the option to suppress the keyinfo certificate

Check?

Verstuurd vanaf mijn iPhone

**Comment by kislyuk** [comment]
If you don't want to include the certificate info in the signed XML, then don't pass the certificate to `sign()`. Just pass the key.

**Comment by kislyuk** [comment]
Actually, after reading the spec a bit more, I think we should use the following API: `xmldsig(data).sign(key=key, key_info=<etree.Element>)`. It should be the responsibility of the caller to prepare the key_info XML element, and `sign()` should verify the passed value against the schema.

**Comment by kislyuk** [comment]
p.s. The KeyInfo element, if passed to `sign()`, would need to be formatted according to http://www.w3.org/TR/xmldsig-core2/#sec-KeyInfo, and would need to be checked against the schema. This link should be included in the corresponding docstring.

**Pull request closed without merging**